# Co-occurrence Clustering Algorithm

One primary reason that makes the analysis of single-cell RNA-seq data challenging is the dropouts, where the data only captures a small fraction of the transcriptome of each cell. Many computational algorithms have been developed to address the dropouts. Here, an opposite view is explored. Instead of treating dropout as a problem to be fixed, we embrace it as a useful signal for defining cell types. We present an iterative co-occurrence clustering algorithm that works with binarized single-cell RNA-seq count data, and is able to effectively identify cell populations, as well as cell-type specific pathways and signatures. 

# User Instructions

Download and unzip the cooccurence_clustering_repo.zip file. Among the three resulting folders, source code is under the "tools" folder, and the other two folders are two example datasets. 

To test the algorithm on the provided examples: open Matlab, change working directory to one example folder, run the step_01, 02, 03, ... scripts sequentially. 

In addition to the co-occurrence clustering algorthim, we also provide a matlab implementation of Seurat clustering.

System requirements for running the code:  Matlab 2017b and Windonws 10 

# Examples

In each example folder, there is a subfolder called html, which contains all the intermediate figures generated by the co-occurrence clustering algorithm. Quick summaries of the examples are shown here. 

### Peripheral Blood Mononuclear Cells (PBMC)

Data Source: The PBMC is available from 10X Genomics (https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz).

Co-occurrence clustering result:
<img src="https://github.com/pqiu/cooccurrence_clustering/blob/master/example_PBMC.PNG" width="800">

### Mouse inner ear sensory epithelia

Data Source: GSE71982

Co-occurrence clustering result:
<img src="https://github.com/pqiu/cooccurrence_clustering/blob/master/example_MouseInnerEar.PNG" width="800">

